addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,255 @@
+
+        self.message_post(body=_("The invoice has been successfully sent to Nilvera."))
+
+    def _l10n_tr_nilvera_post_series(self, endpoint, client):
+        """Post the series to Nilvera based on the endpoint."""
+        if "einvoice" in endpoint:
+            series_endpoint = "/einvoice/Series"
+        elif "earchive" in endpoint:
+            series_endpoint = "/earchive/Series"
+        else:
+            # The endpoint should contain either "einvoice" or "earchive, return early if it's not valid.
+            return
+
+        if not self.sequence_prefix:
+            return
+
+        series = self.sequence_prefix.split('/')[0]
+        client.request(
+            "POST",
+            series_endpoint,
+            json={
+                'Name': series,
+                'IsActive': True,
+                'IsDefault': False,
+            },
+        )
+
+    def _l10n_tr_nilvera_get_submitted_document_status(self):
+        client = _get_nilvera_client(self.env.company)
+        for invoice in self:
+            response = client.request(
+                "GET",
+                f"/einvoice/sale/{invoice.l10n_tr_nilvera_uuid}/Status",
+            )
+
+            nilvera_status = response.get('InvoiceStatus', {}).get('Code')
+            if nilvera_status in dict(invoice._fields['l10n_tr_nilvera_send_status'].selection):
+                invoice.l10n_tr_nilvera_send_status = nilvera_status
+                if nilvera_status == 'error':
+                    invoice.message_post(
+                        body=Markup(
+                            "%s<br/>%s - %s<br/>"
+                        ) % (
+                            _("The invoice couldn't be sent to the recipient."),
+                            response['InvoiceStatus'].get('Description'),
+                            response['InvoiceStatus'].get('DetailDescription'),
+                        )
+                    )
+            else:
+                invoice.message_post(body=_("The invoice status couldn't be retrieved from Nilvera."))
+
+    def _l10n_tr_nilvera_get_documents(self):
+        client = _get_nilvera_client(self.env.company)
+        response = client.request(
+            "GET",
+            "/einvoice/Purchase",
+        )
+
+        if not response.get('Content'):
+            return
+
+        journal = self.env.company.l10n_tr_nilvera_purchase_journal_id
+        if not journal:
+            journal = self.env['account.journal'].search([
+                *self.env['account.journal']._check_company_domain(self.env.company),
+                ('type', '=', 'purchase'),
+            ], limit=1)
+
+        document_uuids = [content.get('UUID') for content in response.get('Content')]
+        document_uuids_count = dict(self.env['account.move']._read_group(
+            [('l10n_tr_nilvera_uuid', 'in', document_uuids)],
+            groupby=['l10n_tr_nilvera_uuid'],
+            aggregates=['__count'],
+        ))
```

Comment:
Just a question about this, as we're doing a groupby on the sql just to get the dict.
Are the uuids gotten from nilvera ever acknowledged/removed or could we get 1M uuids after 3 years of using it?
I'm asking because you're checking if you already received one.

Reply:
![image](https://github.com/user-attachments/assets/6f7c5c45-e3f9-4c17-add3-a6ed09310356)
This was Nilvera's response when I asked if they acknowledge received invoices. I went with the second solution as it meant less API calls. Not sure about your question though, or how they decide which invoices are sent. I can tell you for sure that I'm not getting old invoices. Maybe they send invoices in a time window? I'll ask and get back to you